spec/utility/bulk_queue_spec.rb

```diff
@@ -0,0 +1,159 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'utility/bulk_queue'
+
+describe Utility::BulkQueue do
+  let(:subject) { described_class.new(count_threshold, size_threshold) }
+  let(:count_threshold) { 50 }
+  let(:size_threshold) { 999999999 } # super high default
+
+  context '.add' do
+    before(:each) do
+      allow(subject).to receive(:will_fit?)
+        .and_return(true)
+    end
+
+    context 'when new element won\'t fit any more' do
+      before(:each) do
+        allow(subject).to receive(:will_fit?)
+          .and_return(false)
+      end
+
+      it 'raises an error' do
+        expect { subject.add('some-op') }.to raise_error(Utility::BulkQueue::QueueOverflowError)
+      end
+    end
+
+    context 'when only operation is added' do
+      let(:op) { 'do: something' }
+
+      it 'adds only the operation to the buffer' do
+        subject.add(op)
+
+        expect(subject.pop_all).to include(op)
+      end
+    end
+
+    context 'when operation with payload is added' do
+      let(:op) { 'index: id=12' }
+      let(:payload) { 'text: something, counter: 15' }
+
+      it 'adds both operation and payload to the buffer' do
+        subject.add(op, payload)
+
+        buffer = subject.pop_all
+
+        expect(buffer).to include(op)
+        expect(buffer).to include(payload)
+      end
+    end
+  end
+
+  context '.will_fit?' do
```

Comment:
[nitpick] It's better to use #describe when you start testing a method.

refs: https://www.betterspecs.org/